tracks/rust/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,140 @@
+```
+
+If they don't use `match`:
+```
+Check out using `match` instead of `if`. For some things, like destructuring, it's required. It's an idiom specific to Rust, so I always default to it unless the logic is comparing many different things and a match in that case makes it messy.
+```
+
+If they use an `if` on their `.is_empty()` check:
+```
+See about doing a  `match`, instead of `if`, on `your_trimmed_variable.is_empty()`. And, if it's `true`, instantly output our "Fine. Be that way!" constant. Then, everything else will be in the `false` arm.
+Like this:
+    ```rust
+    match your_trimmed_variable.is_empty() {
+        true => FINE,
+        false => {
+    ```
+Be aware that if you use Clippy, it will tell you that it is improper, it's purely a stylistic choice :)
+We can tell Clippy to allow it with `#[allow(clippy::match_bool)]`.
+```
+
+If they are struggling with how to check if it's a question:
+```
+Have a look at `.ends_with()` to check if it's a question.
+```
+
+If they use regex:
+```
+This is good! Great idea using the regex crate! Let's look at how to do this with just the standard library :)
+```
+Then note the following:
+```
+See about finding a way to check if it's alphabetic without using regex.
+```
+
+If they use separate functions:
+```
+We don't need to create separate functions here. We can put this all in one by setting these calls to variables instead.
+```
+
+If they don't use a `match` to compare the conditions:
+```
+We can `match` on just two conditions: `is_question` and `is_yelling`. Like this:
+    ```rust
+    match (is_question, is_yelling) {
+        (true, false) => SURE,
+        ...
+    ```
+```
+
+If they put their constants inside the function:
+```
+Note that it's common convention to put constants at the beginning of a file like I've done above. If it's a public constant, which is typical, we put it following our `use` lines, along with `struct`s and `enum`s. If it's a private constant, we put it at the beginning of the scope of the function. But, either way doesn't affect the performance or functionality!
+```
+
+If they ask about `&str` having a static lifetime in comparison to using constants:
+```
+To quote from [Rust By Example](https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html): "When `static_string` goes out of scope, the reference can no longer be used, but the data remains in the binary."
+
+What I think the compiler is doing is recreating the pointer to the static data if we use a plain `let` line, whereas with `const`, the data and the pointer are always kept alive.
+
+Also, the use of `const` allows us to have it globally scopable.
+```
+
+If they use `_` instead of `false` in their match statement:
+```
+Just a stylistic note, the `_` in the true/false `match` can simply be `false` because there is no other option :)
```

Comment:
Imho that's not only stylistic. It suggests to the reader that there's something else to catch while there isn't, making it less readable. In other circumstances, it can even be a source of bugs, so the use of `_` should be limited to the cases where it's needed. I'd strengthen this.

Reply:
Great, thanks! I've added more to the suggestion :)